chia/wallet/trade_manager.py

```diff
@@ -391,6 +391,16 @@ async def cancel_pending_offers(
 
     async def save_trade(self, trade: TradeRecord, offer_name: bytes32) -> None:
         await self.trade_store.add_trade_record(trade, offer_name)
+        offer: Offer = Offer.from_bytes(trade.offer)
```

Comment:
This is kind of a bit of wallet magic that I'm not able to fully explain.  You can see here: https://github.com/Chia-Network/chia-blockchain/blob/4603b9d7688030f2cf300140c9592bd003647978/chia/wallet/wallet_state_manager.py#L1545-L1547 that we do this for regular transactions.  If you don't subscribe to the additions and removals, you start getting balance failures in the tests.  It's possible this is a hack, but it predates my involvement in this code so it's a hack that's been around a long time.  Regardless, not something I intend to diagnose and fix in this PR.

Reply:
I did make it so that the offer gets passed through to the save_trade function though

Reply:
> that we do this for regular transactions. If you don't subscribe to the additions and removals, you start getting balance failures in the tests. It's possible this is a hack, but it predates my involvement in this code so it's a hack that's been around a long time. Regardless, not something I intend to diagnose and fix in this PR.

I think it's always important to understand why you add a change to the code, especially if maybe adds more subscriptions for no reason, and not just do something without understanding why you do it only to make tests happy, even if this requires diagnosing an issue or hack which predates your time. 

> I did make it so that the offer gets passed through to the save_trade function though

Thanks!

Reply:
> I think it's always important to understand why you add a change to the code, especially if maybe adds more subscriptions for no reason, and not just do something without understanding why you do it only to make tests happy, even if this requires diagnosing an issue or hack which predates your time

Okay I dove into this, and the reason we do this is because basically _removals_ don't have hints.  The flow is as follows:
1) We receive an unspent coin because we have subscribed to its hint
2) We spend that coin and subscribe to its ID as we push the transaction (the behavior that was in question here)
3) We pick up the coin state for the coin in its unspent state via the ID that we just subscribed to

If we didn't subscribe to all of the removals, we would never pick up the coin state with a `spent_height is not None` and handle it accordingly.